subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultModuleIdentifier.java

```diff
@@ -73,6 +73,6 @@ public boolean equals(Object obj) {
 
     @Override
     public int hashCode() {
-        return 31 * group.hashCode() ^ name.hashCode();
+        return 31 * name.hashCode() ^ group.hashCode();
```

Comment:
It's explained in the commit message: the name is usually shorter than the group, so its hashcode is often within the same range because it won't overflow an int. This change improves the distribution of the hashcode.